### PR TITLE
Enrich instructions for string functions HEX() and INSERT()

### PR DESCRIPTION
**Pull request by Takashi-kun**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

Fix https://github.com/pingcap/docs/issues/15488

This PR adds instructions `HEX()` and `INSERT()`

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [ ] v7.6 (TiDB 7.6 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.4 (TiDB 7.4 versions)
- [ ] v7.3 (TiDB 7.3 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: functions-and-operators/string-functions.md [hunk 1]
```diff
@@ -156,10 +156,92 @@ Decode to a base-64 string and return result.
 ### [`HEX()`](https://dev.mysql.com/doc/refman/8.0/en/string-functions.html#function_hex)
 
 Return a hexadecimal representation of a decimal or string value.
+- For a string argument `HEX(str)` returns a hexademical string representation of `str` where each byte of each character in `str` is converted to two hexadecimal digits.
```
Comment (dveeden) [inline functions-and-operators/string-functions.md:159]:
```suggestion
- For a string argument `HEX(str)` returns a hexadecimal string representation of `str` where each byte of each character in `str` is converted to two hexadecimal digits. For example `a` in a UTF-8 or ASCII character set is encoded as a byte with a binary value of 00111101 or 61 in hexadecimal notation.
```
Feel free to include this or leave this out.

File: functions-and-operators/string-functions.md [hunk 2]
```diff
@@ -156,10 +156,92 @@ Decode to a base-64 string and return result.
...
+- The inverse of this operation is performed by the [ `UNHEX()` ](https://docs.pingcap.com/tidb/dev/string-functions#unhex) function.
```
Comment (dveeden) [inline functions-and-operators/string-functions.md:160]:
```suggestion
- The inverse of this operation is performed by the [`UNHEX()`](/functions-and-operators/string-functions.md#unhex) function.
```

File: functions-and-operators/string-functions.md [hunk 3]
```diff
@@ -156,10 +156,92 @@ Decode to a base-64 string and return result.
...
+- For a numeric argument `HEX(n)` returns a hexadecimal string representation of the value of `n` treated as a longlong (`BIGINT`) number. This is equivalent to [ `CONV(n, 10, 16)` ](https://docs.pingcap.com/tidb/stable/numeric-functions-and-operators#mathematical-functions).
```
Comment (dveeden) [inline functions-and-operators/string-functions.md:161]:
```suggestion
- For a numeric argument `HEX(n)` returns a hexadecimal string representation of the value of `n` treated as a longlong (`BIGINT`) number. This is equivalent to [ `CONV(n, 10, 16)` ](/functions-and-operators/numeric-functions-and-operators#mathematical-functions).
```

File: functions-and-operators/string-functions.md [hunk 4]
```diff
@@ -156,10 +156,92 @@ Decode to a base-64 string and return result.
...
+- For a `NULL` argument, this function returns `NULL`.
+
+```sql
+SELECT X'616263', HEX('abc'), UNHEX(HEX('abc'));
```
Comment (dveeden) [inline functions-and-operators/string-functions.md:165]: Maybe add `0x616263`?
```
mysql> SELECT X'616263', HEX('abc'), UNHEX(HEX('abc')), 0x616263;
+-----------+------------+-------------------+----------+
| X'616263' | HEX('abc') | UNHEX(HEX('abc')) | 0x616263 |
+-----------+------------+-------------------+----------+
| abc       | 616263     | abc               | abc      |
+-----------+------------+-------------------+----------+
1 row in set (0.00 sec)
```

Comment (Takashi-kun) [reply functions-and-operators/string-functions.md]: Nice!
I add it as you said > https://github.com/pingcap/docs/pull/16114/commits/f4d74302a4857828117439221e4a7d38e264d632

**Review by dveeden — APPROVED** [review]

**Review by SeaRise — APPROVED** [review]
lgtm

File: functions-and-operators/string-functions.md [hunk 5]
```diff
@@ -156,10 +156,92 @@ Decode to a base-64 string and return result.
 ### [`HEX()`](https://dev.mysql.com/doc/refman/8.0/en/string-functions.html#function_hex)
 
 Return a hexadecimal representation of a decimal or string value.
+- For a string argument `HEX(str)` returns a hexadecimal string representation of `str` where each byte of each character in `str` is converted to two hexadecimal digits. For example `a` in a UTF-8 or ASCII character set is encoded as a byte with a binary value of 00111101 or 61 in hexadecimal notation.
```
Comment (dveeden) [inline functions-and-operators/string-functions.md:159]:
```suggestion

- For a string argument `HEX(str)` returns a hexadecimal string representation of `str` where each byte of each character in `str` is converted to two hexadecimal digits. For example `a` in a UTF-8 or ASCII character set is encoded as a byte with a binary value of 00111101 or 61 in hexadecimal notation.
```

From the linter:
```
[2024-01-11T09:00:57.576Z] functions-and-operators/string-functions.md: this file has manual line breaks in the following lines:
[2024-01-11T09:00:57.576Z] 
[2024-01-11T09:00:57.576Z] MANUAL LINE BREAKS: L159
[2024-01-11T09:00:57.576Z] MANUAL LINE BREAKS: L197
[2024-01-11T09:00:57.576Z] MANUAL LINE BREAKS: L198
```

File: functions-and-operators/string-functions.md [hunk 6]
```diff
@@ -156,10 +156,92 @@ Decode to a base-64 string and return result.
...
+- The inverse of this operation is performed by the [`UNHEX()`](/functions-and-operators/string-functions.md#unhex) function.
+- For a numeric argument `HEX(n)` returns a hexadecimal string representation of the value of `n` treated as a longlong (`BIGINT`) number. This is equivalent to [ `CONV(n, 10, 16)` ](/functions-and-operators/numeric-functions-and-operators#mathematical-functions).
```
Comment (dveeden) [inline functions-and-operators/string-functions.md:161]:
```suggestion
- For a numeric argument `HEX(n)` returns a hexadecimal string representation of the value of `n` treated as a longlong (`BIGINT`) number. This is equivalent to [`CONV(n, 10, 16)`](/functions-and-operators/numeric-functions-and-operators#mathematical-functions).
```

From the linter:
```
[2024-01-11T09:00:59.985Z] functions-and-operators/string-functions.md: 159: MD032/blanks-around-lists Lists should be surrounded by blank lines [Context: "- For a string argument `HEX(s..."]
[2024-01-11T09:00:59.985Z] functions-and-operators/string-functions.md: 161: MD039/no-space-in-links Spaces inside link text [Context: "[ CONV(n, 10, 16) ]"]
[2024-01-11T09:00:59.985Z] functions-and-operators/string-functions.md: 198: MD032/blanks-around-lists Lists should be surrounded by blank lines [Context: "- Returs the original string `..."]
```
```

**Review by dveeden — APPROVED** [review]

File: functions-and-operators/string-functions.md [hunk 7]
```diff
@@ -157,9 +157,93 @@ Decode to a base-64 string and return result.
...
+| あいいいあああ                                     |
++----------------------------------------------------+
+
+SELECT INSERT('あああああああ', 2, 3, 'xx');
++---------------------------------------------+
+| INSERT('あああああああ', 2, 3, 'xx')        |
++---------------------------------------------+
+| あxxあああ                                  |
++---------------------------------------------+
+```
```
Comment (Oreoxmt) [inline functions-and-operators/string-functions.md:246]:
````suggestion
Examples:

```sql
SELECT INSERT('He likes tennis', 4, 5, 'plays');
+------------------------------------------+
| INSERT('He likes tennis', 4, 5, 'plays') |
+------------------------------------------+
| He plays tennis                          |
+------------------------------------------+
```

```sql
SELECT INSERT('He likes tennis', -1, 5, 'plays');
+-------------------------------------------+
| INSERT('He likes tennis', -1, 5, 'plays') |
+-------------------------------------------+
| He likes tennis                           |
+-------------------------------------------+
```

```sql
SELECT INSERT('He likes tennis', 4, 100, 'plays');
+--------------------------------------------+
| INSERT('He likes tennis', 4, 100, 'plays') |
+--------------------------------------------+
| He plays                                   |
+--------------------------------------------+
```

```sql
SELECT INSERT('He likes tenis', 10, 100, '🍣');
+-------------------------------------------+
| INSERT('He likes tenis', 10, 100, '🍣')     |
+-------------------------------------------+
| He likes 🍣                                 |
+-------------------------------------------+
```

```sql
SELECT INSERT('あああああああ', 2, 3, 'いいい');
+----------------------------------------------------+
| INSERT('あああああああ', 2, 3, 'いいい')           |
+----------------------------------------------------+
| あいいいあああ                                     |
+----------------------------------------------------+
```

```sql
SELECT INSERT('あああああああ', 2, 3, 'xx');
+---------------------------------------------+
| INSERT('あああああああ', 2, 3, 'xx')        |
+---------------------------------------------+
| あxxあああ                                  |
+---------------------------------------------+
```
````

File: functions-and-operators/string-functions.md [hunk 8]
```diff
@@ -157,9 +157,93 @@ Decode to a base-64 string and return result.
...
+| FF       | 255                    |
++----------+------------------------+
+
+SELECT HEX(NULL);
++-----------+
+| HEX(NULL) |
++-----------+
+| NULL      |
++-----------+
+```
```
Comment (Oreoxmt) [inline functions-and-operators/string-functions.md:193]:
````suggestion
Examples:

```sql
SELECT X'616263', HEX('abc'), UNHEX(HEX('abc')), 0x616263;
+-----------+------------+-------------------+----------+
| X'616263' | HEX('abc') | UNHEX(HEX('abc')) | 0x616263 |
+-----------+------------+-------------------+----------+
| abc       | 616263     | abc               | abc      |
+-----------+------------+-------------------+----------+
```

```sql
SELECT X'F09F8DA3', HEX('🍣'), UNHEX(HEX('🍣')), 0xF09F8DA3;
+-------------+-------------+--------------------+------------+
| X'F09F8DA3' | HEX('🍣')     | UNHEX(HEX('🍣'))     | 0xF09F8DA3 |
+-------------+-------------+--------------------+------------+
| 🍣            | F09F8DA3    | 🍣                   | 🍣           |
+-------------+-------------+--------------------+------------+
```

```sql
SELECT HEX(255), CONV(HEX(255), 16, 10);
+----------+------------------------+
| HEX(255) | CONV(HEX(255), 16, 10) |
+----------+------------------------+
| FF       | 255                    |
+----------+------------------------+
```

```sql
SELECT HEX(NULL);
+-----------+
| HEX(NULL) |
+-----------+
| NULL      |
+-----------+
```
````

File: functions-and-operators/string-functions.md [hunk 9]
```diff
@@ -157,9 +157,93 @@ Decode to a base-64 string and return result.
 
 Return a hexadecimal representation of a decimal or string value.
 
+- For a string argument `HEX(str)` returns a hexadecimal string representation of `str` where each byte of each character in `str` is converted to two hexadecimal digits. For example `a` in a UTF-8 or ASCII character set is encoded as a byte with a binary value of 00111101 or 61 in hexadecimal notation.
+- The inverse of this operation is performed by the [`UNHEX()`](/functions-and-operators/string-functions.md#unhex) function.
+- For a numeric argument `HEX(n)` returns a hexadecimal string representation of the value of `n` treated as a longlong (`BIGINT`) number. This is equivalent to [`CONV(n, 10, 16)`](/functions-and-operators/numeric-functions-and-operators#mathematical-functions).
+- For a `NULL` argument, this function returns `NULL`.
```
Comment (Oreoxmt) [inline functions-and-operators/string-functions.md:163]:
```suggestion
The `HEX()` function is used to convert the given argument into a string representation of its hexadecimal value. The argument can be either a string or a number.

- If the argument is a string, `HEX(str)` returns a hexadecimal string representation of `str`. The function converts each byte of each character in `str` into two hexadecimal digits. For example, the character `a` in a UTF-8 or ASCII character set is represented as a binary value of `00111101`, or `61` in hexadecimal notation.
- If the argument is a number, `HEX(n)` returns a hexadecimal string representation of `n`. The function treats the argument `n` as a `BIGINT` number, equivalent to using `CONV(n, 10, 16)`.
- If the argument is `NULL`, the function returns `NULL`.
```

File: functions-and-operators/string-functions.md [hunk 10]
```diff
@@ -157,9 +157,93 @@ Decode to a base-64 string and return result.
...
+
 ### [`INSERT()`](https://dev.mysql.com/doc/refman/8.0/en/string-functions.html#function_insert)
 
-Insert a substring at the specified position up to the specified number of characters.
+`INSERT(str, pos, len, newstr)` returns the string `str` with the substring beginning at position `pos` and `len` characters long which is replaced by `newstr`.
+
+This function is multibyte safe.
+- Returs the original string `str` if `pos` is not within the length of string,
+- Returns the string which is replaced the rest of the string from position `pos` if `len` is not within the length of the rest of string.
+- Returns `NULL` if any argument is `NULL`.
```
Comment (Oreoxmt) [inline functions-and-operators/string-functions.md:202]:
```suggestion
The `INSERT(str, pos, len, newstr)` function is used to replace a substring in `str` that starts at position `pos` and is `len` characters long with the string `newstr`. This function is multibyte safe.

- If `pos` exceeds the length of `str`, the function returns the original string `str` without modification.
- If `len` exceeds the remaining length of `str` from position `pos`, the function replaces the rest of the string from position `pos`.
- If any argument is `NULL`, the function returns `NULL`.
```

**Comment by Oreoxmt** [comment]
Hi @Takashi-kun Thanks for your contribution. I've added a few comments. PTAL

**Comment by Takashi-kun** [comment]
@Oreoxmt Thanks for your review!
I confirmed and merged your suggestions all, please take a look them again 🙇

**Review by Oreoxmt — APPROVED** [review]

File: functions-and-operators/string-functions.md [hunk 11]
```diff
@@ -155,11 +155,113 @@ Decode to a base-64 string and return result.
...
+| HEX(NULL) |
++-----------+
+| NULL      |
++-----------+
+```
 
 ### [`INSERT()`](https://dev.mysql.com/doc/refman/8.0/en/string-functions.html#function_insert)
 
-Insert a substring at the specified position up to the specified number of characters.
+The `INSERT(str, pos, len, newstr)` function is used to replace a substring in `str` that starts at position `pos` and is `len` characters long with the string `newstr`. This function is multibyte safe.
```
Comment (qiancai) [inline functions-and-operators/string-functions.md:204]:
```suggestion
The `INSERT(str, pos, len, newstr)` function is used to replace a substring in `str` (that starts at position `pos` and is `len` characters long) with the string `newstr`. This function is multibyte safe.
```

**Review by qiancai — APPROVED** [review]
Rest LGTM

**Pull request merged**